### PR TITLE
feat: Add batch replayer

### PR DESCRIPTION
**Pull request by ijemmy**
Issue #, if available:

Example usage:
```
const integTestBucket = new Bucket(testStack, "IntegTestBucket");

new BatchReplayer(testStack, "CoreIntegBatchReplayer", {
  dataset: PartitionedDataset.RETAIL_1GB_WEB_SALE,
  frequency: 600, 
  sinkBucket: integTestBucket,
  outputFileMaxSizeInBytes: 20480, // Default to 100MB
});
```

Description of changes:
`BatchReplayer` is similar to the current (batch) data generator, but with these difference:
1. It uses `PartitionedDataset` instead of `Dataset`. This type of dataset has been partitioned and has a manifest file. This makes reading data from the source bucket much more efficient.
2. Instead of using Athena which we have no coding control, we use a step function with two lambda functions. The first function read the dataset's manifest file and pass an array of files within range to the 2nd function. The 2nd function parse each file (using "Map" state in step function), filter, adjust date columns, and write to`sinkBucket`.
3. It has an optional prop `outputFileMaxSizeInBytes`. This will limit a file size that's written to the `sinkBucket`. For example, if the dataframe to write is 100MB but this value is set to 1MB. It will split records into 101 files. Each of these files won't exceed 1MB.  (Not 100 file as we have to duplicate headers for each file).

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: core/src/data-generator/batch-replayer.ts [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+      // Retry on 500 error on invocation with an interval of 2 sec with back-off rate 2, for 6 times
+      retryOnServiceExceptions: true,
+      outputPath: "$.Payload",
+    });
+
+
+    /**
+     * Write data in batch step
+     */
+    const writeInBatchFn = new lambda.DockerImageFunction(this, "writeInBatchFn", {
```
Comment (vgkowski) [inline core/src/data-generator/batch-replayer.ts:114]: can we use PreBundledFunction?

File: core/src/data-generator/batch-replayer.ts [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+    this.dataset = props.dataset;
+    this.frequency = props.frequency || 60;
+    this.sinkBucket = props.sinkBucket;
+    this.outputFileMaxSizeInBytes = props.outputFileMaxSizeInBytes || 100 * 1024 * 1024; //Default to 100 MB
+    
+
+    /**
+     * Find all paths within the time range from the manifest file
+     */
+    const findFilePathsFn = new lambda.DockerImageFunction(this, "findFilePathFn", {
```
Comment (vgkowski) [inline core/src/data-generator/batch-replayer.ts:74]: can we use PreBundledFunction?

File: core/src/data-generator/batch-replayer.ts (hunk 1 above)
Comment (ijemmy) [reply core/src/data-generator/batch-replayer.ts:114]: No, it exceeds Lambda package size. I was considering a Lambda layer but there is no official one.

File: core/src/data-generator/batch-replayer.ts (hunk 2 above)
Comment (ijemmy) [reply core/src/data-generator/batch-replayer.ts:74]: No, it exceeds Lambda package size. I was considering a Lambda layer but there is no official one.

**Review by vgkowski — DISMISSED** [review]

**Review by vgkowski — APPROVED** [review]

**Pull request merged**